substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/oldobject/TestArrayLeak.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2023, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.test.jfr.oldobject;
+
+import jdk.jfr.Recording;
+import org.junit.Test;
+
+public class TestArrayLeak extends JfrOldObjectTest {
+    @Test
+    public void testArrayLeak() throws Throwable {
+        Recording recording = startRecording();
+
+        Object[] node = new Object[3];
+        leak = node;
+        for (int i = 0; i < 1_000_000; i++) {
+            Object[] value = new Object[100];
+            node[0] = value;
+            Object[] left = new Object[3];
+            node[1] = left;
+            Object[] right = new Object[3];
+            node[2] = right;
+            node = right;
+        }
```

Comment:
> Is this adding objects to this array try to make the 100 element array allocation get sampled along the allocation slow path? If so, is there a chance that it doesn't actually get sampled?

All I'm doing is generating enough arrays so that some of them get sampled as old objects and I can verify that sampled arrays make it through as events and they have the right data populated. Having the right data here means that the sampled event is for Object[] has length of 100. If no arrays were sampled the test would fail because it guarantees that the list of events generated for the `Object[]` is not empty.

> What if you directly called JfrOldObjectSampleEvents.sample(result, size.rawValue(), Integer.MIN_VALUE); from this testing code?

Hmmm, not sure what that would do that this test is not already covering?